### ISSUE TITLE
Need CoCreateInstance()

### ISSUE THREAD

**Issue by rodrigc** [opalmer/pywincffi#127]
In Twisted, I tried to replace pywin32 with pywincffi: https://github.com/twisted/twisted/pull/920

I found that pythoncom.CoCreateInstance() is not in pywincffi.

@opalmer can you add it?

**Comment by opalmer** [comment]
It's a standard win32 function supported since Windows 2000 so it should be possible. Based on the function signature and what it's supposed to do this will probably require a bit more work than #123 / #124 did. 

The way I do testing on pywincffi is to actually utilize the functions in a way that you might use them in code. Since I don't have anything related to `CoCreateInstance` yet I may have to add some additional functions just to make sure I can properly test the change. On the bright side this looks like it _might_ be related to some of the things `PumpWaitingMessages()` might do (based on the docs) so this and #122 may be workable at the same time.

**Comment by exvito** [comment]
For consideration / discussion,

It seems `CoCreateInstance` is used by Twisted to open Windows shell shortcuts via `twisted.python.shortcut.open`. Other than being a public Twisted API, it does not seem to be used anywhere within Twisted itself.

Given that implementing `CoCreateInstance` will require bringing in what looks like a non-trivial amount of Windows COM related complexity to pywincffi, would it be acceptable to consider the option of deprecating such an API from Twisted itself?

It certainly feels out of place within Twisted.

Maybe this discussion should be brought to the Twisted mailing list, what do you say @rodrigc?

**Comment by opalmer** [comment]
Agreed on my end...not because it's less work but because this seems like the most logical approach in my opinion. The same thing was done for registry lookups although in that case Python's standard library does provide an equivalent API.
